core/src/main/scala/chisel3/properties/Property.scala

```diff
@@ -14,14 +15,31 @@ import scala.annotation.implicitNotFound
   * Chisel.
   */
 @implicitNotFound("unsupported Property type ${T}")
-private[chisel3] class PropertyType[T]
+private[chisel3] trait PropertyType[T] {
+
+  /** Get the IR PropertyType for this PropertyType.
+    */
+  def getPropertyType(): ir.PropertyType
```

Comment:
Kind of a nit: Scala coding conventions treat a 0-arity function declared with parentheses as a function which _has side effects_ and one that does not have parentheses as one which _does not have side effects_. This is admittedly weird. Given that, this should be declared without parentheses.

Reply:
ah i was not aware of this convention, thanks!